app/auth/middleware.js

```diff
@@ -15,20 +11,16 @@ function processAuthResponse(defaultLocations = []) {
     }
 
     try {
-      const accessToken = _decodeAccessToken(grant.response.access_token)
-      const locations = await referenceData.getLocationsById(defaultLocations)
-
       req.session.regenerate(error => {
         if (error) {
           return next(error)
         }
 
-        req.session.authExpiry = accessToken.exp
-        req.session.postAuthRedirect = postAuthRedirect
-        req.session.user = new User({
-          ...accessToken,
-          locations,
-        })
+        _setSessionData(
```

Comment:
I'd probably just scrap this. Is it just to beat code climate's warning of line length? If it's just that we can manually ignore that warning for this. I don't think splitting this out helps with re-use or readability to be honest.

If you were to refactor at all it would be neater to separate out the `session.regenerate` callback rather than have half the function inline and the other half separated.

Reply:
I agree it's not any better. I'm not sure it's possible to extract the callback function easily because it'll need scope to access `req` and `next` which are scoped to the parent `middleware` function. Sure it's possible but it's going to be a meaty refactoring (unless you can think of an easy way). Are you happy for me to revert it back and mark it as won't fix?

Reply:
Yeah, given that would be a little messier too I'd just revert to what it was before and we can mark code climate to ignore the line length in this particular instance.